src/host.js

```diff
@@ -10,9 +10,12 @@ export default class Host extends Port {
 			throw new Error(`Unable to extract origin from "${src}"`);
 		}
 
-		var parent = document.getElementById(id);
-		if(parent === null) {
-			throw new Error(`Could not find parent node with id "${id}"`);
+		var parent = id;
+		if (typeof id === 'string') {
```

Comment:
I hope they have it? If not it's always something we could add later -- call the function and check if it's a reference to a promise.